src/java/org/apache/cassandra/net/ParamType.java

```diff
@@ -40,25 +39,17 @@
  * will skip over any params it doesn't recognise.
  *
  * Please don't add boolean params here. Extend and use {@link MessageFlag} instead.
+ *
+ * Do not re-use old, nor fill gaps in the sequence of, ids.  New IDs must be higher.
```

Comment:
`ParamType#lookUpByAlias` is not used anymore, so we can get rid of it. With this method gone, `ParamType#legacyAlias` and `ParamType.aliasToTypeMap` can also be removed, and the constructor can take only the id and the serializer.